core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -720,11 +729,17 @@ private void cleanupSymbolMapFiles(Path path, int rootLen, CharSequence columnNa
     }
 
     private void closeSegmentSwitchFiles(LongList newColumnFiles) {
+        int commitMode = configuration.getCommitMode();
+
         // Each record is about primary and secondary file. File descriptor is set every half a record.
         int halfRecord = NEW_COL_RECORD_SIZE / 2;
         for (int fdIndex = 0; fdIndex < newColumnFiles.size(); fdIndex += halfRecord) {
             final int fd = (int) newColumnFiles.get(fdIndex);
-            ff.close(fd);
+            if (commitMode != CommitMode.NOSYNC) {
+                ff.fsyncAndClose(fd);
```

Comment:
We should consider commit mode from config here.

Reply:
I don't quite follow, we do act here according to the config.